packages/announcement-controller/src/AnnouncementController.ts

```diff
@@ -1,96 +1,119 @@
+const metadata = {
+  announcements: {
+    persist: true,
+    anonymous: false,
+  },
+};
+
+export type AnnouncementControllerMessenger = RestrictedControllerMessenger<
+  typeof controllerName,
+  AnnouncementControllerActions,
+  AnnouncementControllerEvents,
+  never,
+  never
+>;
+
 /**
  * Controller for managing in-app announcements.
  */
-export class AnnouncementController extends BaseController<
-  AnnouncementConfig,
-  AnnouncementState
+export class AnnouncementController extends BaseControllerV2<
+  typeof controllerName,
+  AnnouncementControllerState,
+  AnnouncementControllerMessenger
 > {
+  // allAnnouncements: StateAnnouncementMap;
   /**
    * Creates a AnnouncementController instance.
    *
-   * @param config - Initial options used to configure this controller.
-   * @param state - Initial state to set on this controller.
+   * @param args - The arguments to this function.
+   * @param args.messenger - Messenger used to communicate with BaseV2 controller.
+   * @param args.state - Initial state to set on this controller.
+   * @param args.allAnnouncements - Announcements to be passed through to #addAnnouncements
    */
-  constructor(config: AnnouncementConfig, state?: AnnouncementState) {
-    super(config, state || defaultState);
-    this.initialize();
-    this._addAnnouncements();
+  constructor({
+    messenger,
+    state,
+    allAnnouncements,
+  }: {
+    messenger: AnnouncementControllerMessenger;
+    state?: AnnouncementControllerState;
+    allAnnouncements: StateAnnouncementMap;
+  }) {
+    const mergedState = { ...defaultState, ...state };
+    super({ messenger, metadata, name: controllerName, state: mergedState });
+    this.#addAnnouncements(allAnnouncements);
   }
 
   /**
    * Compares the announcements in state with the announcements from file
    * to check if there are any new announcements
    * if yes, the new announcement will be added to the state with a flag indicating
    * that the announcement is not seen by the user.
+   *
+   * @param allAnnouncements - all announcements to compare with the announcements from state
    */
-  private _addAnnouncements(): void {
-    const newAnnouncements: StateAnnouncementMap = {};
-    const { allAnnouncements } = this.config;
-    Object.values(allAnnouncements).forEach(
-      (announcement: StateAnnouncement) => {
-        newAnnouncements[announcement.id] = this.state.announcements[
-          announcement.id
-        ]
-          ? this.state.announcements[announcement.id]
-          : {
-              ...announcement,
-              isShown: false,
-            };
-      },
-    );
-    this.update({ announcements: newAnnouncements });
+  #addAnnouncements(allAnnouncements: StateAnnouncementMap): void {
+    this.update(({ announcements }) => {
+      Object.values(allAnnouncements).forEach(
+        (announcement: StateAnnouncement) => {
+          announcements[announcement.id] =
+            allAnnouncements[announcement.id] ?? announcement;
```

Comment:
It looks like this line isn't covered by tests (which is why CI is failing). Can you add a test for this?